terraform/virtual-machine.tf

```diff
@@ -34,49 +42,55 @@ resource "azurerm_linux_virtual_machine" "vh_ado_agent" {
   network_interface_ids = [azurerm_network_interface.vh_ado_agent_nic[each.value.name].id]
   size                  = "Standard_D4s_v3"
 
-  disable_password_authentication = false #tfsec:ignore:azure-compute-disable-password-authentication
-  admin_username                  = var.vm_username
-  admin_password                  = random_password.password.result
+  admin_username = var.vm_username
+  admin_password = random_password.password.result
+
+  identity {
+    type = "SystemAssigned"
+  }
 
   os_disk {
-    name                 = "${each.value.name}-os-disk"
+    name                 = each.value.os_disk
     caching              = "ReadWrite"
     storage_account_type = "Premium_LRS"
     disk_size_gb         = 128
   }
 
   source_image_reference {
-    publisher = "Canonical"
-    offer     = "UbuntuServer"
-    sku       = "18.04-LTS"
-    version   = "latest"
+    publisher = local.publisher
+    offer     = local.offer
+    sku       = local.sku
+    version   = local.version
   }
 
   tags = local.common_tags
 
   depends_on = [
-    azurerm_network_interface.vh_ado_agent_nic
+    azurerm_network_interface.vh_ado_agent_nic,
+    azurerm_automation_account.vh_infra_core_ado
   ]
-}
 
+}
 
-resource "azurerm_virtual_machine_extension" "AzureDevOpsAgent" {
+resource "azurerm_managed_disk" "vh_ado_agent" {
   for_each = local.vms
 
-  name                 = "AzureDevOpsAgent"
-  publisher            = "Microsoft.Azure.Extensions"
-  type                 = "CustomScript"
-  type_handler_version = "2.1"
-  virtual_machine_id   = azurerm_linux_virtual_machine.vh_ado_agent[each.value.name].id
-
-  protected_settings = <<PROTECTED_SETTINGS
-      {
-          "script": "${filebase64("set_up.sh")}"
-      }
-  PROTECTED_SETTINGS
+  name                 = each.value.data_disk
+  location             = azurerm_resource_group.vh_infra_core_ado.location
+  resource_group_name  = azurerm_resource_group.vh_infra_core_ado.name
+  storage_account_type = "Premium_LRS"
+  create_option        = "Empty"
+  disk_size_gb         = "64"
 
   tags = local.common_tags
-  depends_on = [
-    azurerm_linux_virtual_machine.vh_ado_agent
-  ]
+
+}
+
+resource "azurerm_virtual_machine_data_disk_attachment" "data_disk_attachment" {
+  for_each = local.vms
+
+  managed_disk_id    = azurerm_managed_disk.vh_ado_agent[each.value.name].id
+  virtual_machine_id = azurerm_windows_virtual_machine.vh_ado_agent[each.value.name].id
+  lun                = "10"
```

Comment:
why is this lun 10 ?

Reply:
I have just always used multiples of ten - it was just how i was taught, happy to use anything else if required

Reply:
yeah, can we start at 0 please so we follow all other VM's on the platform, never seen it start at 10 before